### PR TITLE
Remove TargetIdGenerator

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This PR removes the TargetIdGenerator, which offered a fancy way to add +2 to integers.

### CODE REVIEW

File: packages/firestore/test/unit/specs/spec_builder.ts [hunk 1]
```diff
@@ -94,7 +92,13 @@ export class ClientMemoryState {
     this.queryMapping = new ObjectMap<Target, TargetId>(t => t.canonicalId());
     this.limboMapping = {};
     this.activeTargets = {};
-    this.limboIdGenerator = TargetIdGenerator.forSyncEngine();
+    this.nextLimboTarget = 1;
+  }
+
+  nextLimboTargetId(): TargetId {
```
Comment (dconeybe) [inline packages/firestore/test/unit/specs/spec_builder.ts]: nit: Naming the `nextLimboTargetId()` method the same as the variable name somewhat implies that it is a simple "getter" method.  Consider renaming to something that implies that it is "doing something" not just "getting something".  e.g. `generateLimboTargetId()`.

File: packages/firestore/test/unit/specs/spec_builder.ts [hunk 2]
```diff
@@ -170,8 +175,8 @@ export class SpecBuilder {
     return this.currentClientState;
   }
 
-  private get limboIdGenerator(): TargetIdGenerator {
-    return this.clientState.limboIdGenerator;
+  private nextLimboTargetId(): TargetId {
```
Comment (dconeybe) [inline packages/firestore/test/unit/specs/spec_builder.ts]: Same here; consider renaming the `nextLimboTargetId()` method.

File: packages/firestore/src/core/target_id_generator.ts [hunk 3]
```diff
@@ -15,83 +15,43 @@
...
-    return this.next();
-  }
-
-  private seek(targetId: TargetId): void {
-    assert(
-      (targetId & RESERVED_BITS) === this.generatorId,
-      'Cannot supply target ID from different generator ID'
-    );
-    this.nextId = targetId;
+    this.lastId += OFFSET;
```
Comment (wilhuff) [inline packages/firestore/src/core/target_id_generator.ts:41]: This is functionally different from what came before. This is now a pre-increment where before it was post-increment. Just double-checking: is this intended?

Comment (schmidt-sebastian) [reply packages/firestore/src/core/target_id_generator.ts:41]: Yes. The target ID generator now gets initialized with 0 and -1 (shudder), which works around this inconsistency.

**Pull request merged**